### PR TITLE
Add ESLint shareable config, remove JShint

### PR DESCRIPTION
**Pull request by omh1280**
Part of https://github.com/AnalyticalGraphicsInc/cesium/pull/5431.

- Add `.eslintrc.json` files in place of `.jshintrc`'s
- Update the `package.json` as appropriate
- Add eslint hints where necessary to make sure `npm run eslint` passes.
- Remove `.idea/jsLinters/jshint.xml`

This is part of an effort to move to [shareable configs](http://eslint.org/docs/developer-guide/shareable-configs): https://www.npmjs.com/package/eslint-config-cesium

### CODE REVIEW

**Comment by omh1280** [comment]
@mramato How does this look? I'll apply the same "treatment" to the other repos if this looks OK.

File: .npmignore [hunk 1]
```diff
@@ -5,7 +5,6 @@
 /output
 /specs
 /test
-.jshintrc
```
Comment (mramato) [inline .npmignore]: We should add the eslint file here as well.

File: index.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/*eslint strict: ["error", "function"]*/
```
Comment (mramato) [inline index.js]: Rather than ignoring the error, just add a `'use strict';` here.

File: lib/removeObject.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ module.exports = removeObject;
  */
 function removeObject(gltf, name, usedId) {
     // Iterate through objects and remove those that are not in the hash
-    var numberRemoved = 0;
+    var numberRemoved = 0; //eslint-disable-line no-unused-vars
```
Comment (mramato) [inline lib/removeObject.js]: Just remove this variable, since it looks like `numberRemoved` isn't ever read.

File: index.js (hunk 2 above)
Comment (mramato) [inline index.js]: Rather than ignoring the error, just add a `'use strict';` here.

Comment (omh1280) [inline index.js]: Got it. For files like `specs/matchers/equals.js` as well?

**Comment by omh1280** [comment]
Bump

File: specs/.eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "../.eslintrc.json",
+  "env": {
+    "jasmine": true
+  },
+  "rules": {
+    "no-unused-vars": "off"
```
Comment (mramato) [inline specs/.eslintrc.json]: Are we sure we want this rule here?

Comment (omh1280) [inline specs/.eslintrc.json]: I copied it from https://github.com/AnalyticalGraphicsInc/gltf-pipeline/pull/288/files/30d340f9d3c9b7d222f59431bd1e11a5f078c952#diff-6deaf0988d1fe3f5d08d3f8bca32b624L4. Want me to turn it off and remove unused vars?

Comment (mramato) [inline specs/.eslintrc.json]: Yes, there are many cases in this repository where variables are really unused and should be removed

In most places where there is a false positive, disable the rule at the line level, except for files like `addDefaultMatchers` where is makes more sense at the file level.

Thanks.

**Comment by omh1280** [comment]
Bump!

File: specs/lib/compressTexturesSpec.js [hunk 5]
```diff
@@ -41,7 +41,7 @@ function compressGltfTexture(gltfPath, imagePath, options) {
                 basePath : basePath
             };
             return Pipeline.processJSON(gltf, pipelineOptions)
-                .then(function(gltf) {
+                .then(function(gltf) { //eslint-disable-line no-unused-vars
```
Comment (mramato) [inline specs/lib/compressTexturesSpec.js]: If `gltf` isn't used here, remove it.

File: specs/lib/removePipelineExtrasSpec.js [hunk 6]
```diff
@@ -7,7 +7,7 @@ describe('removePipelineExtras', function() {
     var gltf;
 
     beforeAll(function(done) {
-        fs.readFile(gltfExtrasPath, function(err, data) {
+        fs.readFile(gltfExtrasPath, function(err, data) { //eslint-disable-line no-unused-vars
```
Comment (mramato) [inline specs/lib/removePipelineExtrasSpec.js]: Remove this comment and fix the function by adding

```javascript
if(defined(err) {
    done.fail(err);
    return;
}
```
to the top of the function.

**Comment by mramato** [comment]
Just those comments and then I'll merge.  Thanks @omh1280!

**Comment by omh1280** [comment]
Thanks! Bump

File: specs/lib/removePipelineExtrasSpec.js [hunk 7]
```diff
@@ -1,13 +1,18 @@
 'use strict';
 var fs = require('fs');
+var defined = require('cesium').defined;
```
Comment (mramato) [inline specs/lib/removePipelineExtrasSpec.js]: Coding guidelines request that this be written as 

```javascript
var Cesium = require('cesium');
var defined = Cesium.defined;
```

Plus, the `var removePipelineExtras = require('../../lib/removePipelineExtras');` should be moved up to be with the other requires, and defined should be after that.

**Comment by omh1280** [comment]
thanks. Bump!

**Comment by mramato** [comment]
Thanks.

**Pull request merged**